core/src/main/java/org/hisp/dhis/android/core/arch/api/internal/PreventURLDecodeInterceptor.java

```diff
@@ -26,29 +26,31 @@
  *  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-package org.hisp.dhis.android.core.arch.api.internal
+package org.hisp.dhis.android.core.arch.api.internal;
 
-import io.ktor.client.plugins.api.createClientPlugin
-import io.ktor.client.request.HttpRequestBuilder
-import io.ktor.http.takeFrom
+import java.io.IOException;
 
-internal object PreventURLDecodePlugin {
-    val instance = createClientPlugin(name = "PreventURLDecodePlugin") {
-        onRequest { request, _ ->
-            replaceEncodedCharacters(request)
-        }
-    }
+import okhttp3.Interceptor;
+import okhttp3.Request;
+import okhttp3.Response;
+
+public class PreventURLDecodeInterceptor implements Interceptor {
+
+    @Override
+    public Response intercept(Chain chain) throws IOException {
+        Request request = chain.request();
+        String encodedUrl = request.url().toString();
 
-    private fun replaceEncodedCharacters(request: HttpRequestBuilder) {
-        val encodedUrl = request.url.toString()
+        String nonEncodedUrl = encodedUrl
```

Comment:
Maybe is not valuable, but in order to escalate replacements you can do a map of replacements and then a fold:

`val replacements = mapOf(
    "%2C" to ",",
    "%5B" to "[",
    "%5D" to "]",
    "%3A" to ":"
)`

so you can add more replacements in the future, and then:
`val nonEncodedUrl = replacements.entries.fold(encodedUrl) { acc, (encoded, decoded) ->
    acc.replace(encoded, decoded)
}`

also you could create an extension function